src/Bridge/Doctrine/Orm/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Bridge\Doctrine\Orm\EventListener;
+
+use ApiPlatform\Core\Bridge\Doctrine\Common\EventListener\AbstractPublishMercureUpdatesListener;
+use Doctrine\Common\EventArgs;
+use Doctrine\ORM\Event\OnFlushEventArgs;
+
+/**
+ * Publishes resources updates to the Mercure hub.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ *
+ * @experimental
+ */
+final class PublishMercureUpdatesListener extends AbstractPublishMercureUpdatesListener
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function onFlush(EventArgs $eventArgs): void
+    {
+        if (!$eventArgs instanceof OnFlushEventArgs) {
+            return;
+        }
+
+        $uow = $eventArgs->getEntityManager()->getUnitOfWork();
+
+        foreach ($uow->getScheduledEntityInsertions() as $entity) {
```

Comment:
If the only difference between ODM and ORM is the method name, maybe could we merge all classes and do something like:

```php
$methodName = eventArgs instanceof OdmOnFlushEventArgs ? getScheduledDocumentInsertions : getScheduledEntityInsertions;

foreach ($uow->$methodName() //...
```

WDYT?
```

Reply:
I've no strong opinion about this. Calling a method dynamically is often considered a bad practice. And having a separate class allows to customize things easily without adding conditional statements.
But on the other hand it multiplies the number of classes and adds a complexity by having an inherited abstract class.